In [1]:
import numpy as np
import pandas as pd
from scipy import stats

__Задача 1__

Проведите дисперсионный анализ для определения того, есть ли различия среднего роста среди взрослых футболистов, хоккеистов и штангистов. Даны значения роста в трёх группах случайно выбранных спортсменов: 
```
football_players = [173, 175, 180, 178, 177, 185, 183, 182]
hockey_players = [177, 179, 180, 188, 177, 172, 171, 184, 180]
lifters = [172, 173, 169, 177, 166, 180, 178, 177, 172, 166, 170]
```

In [15]:
fp = np.array([173, 175, 180, 178, 177, 185, 183, 182])
hp= np.array([177, 179, 180, 188, 177, 172, 171, 184, 180])
lt = np.array([172, 173, 169, 177, 166, 180, 178, 177, 172, 166, 170])

n1 = fp.shape[0]
n2 = hp.shape[0]
n3 = lt.shape[0]

n1, n2, n3

(8, 9, 11)

In [11]:
fp_mean = fp.mean()
hp_mean = hp.mean()
lt_mean = lt.mean()
print('Cреднее значение по выборке равно: \n -для футболистов {},\n -для хоккеистов {},\n -для штангистов {}'.format(fp_mean, hp_mean, lt_mean))

Cреднее значение по выборке равно: 
 -для футболистов 179.125,
 -для хоккеистов 178.66666666666666,
 -для штангистов 172.72727272727272


In [13]:
height = np.concatenate([fp, hp, lt])
height_mean = height.mean()
print('Cреднее значение по всем наблюдениям {}'.format(height_mean))

Cреднее значение по всем наблюдениям 176.46428571428572


Найдём значения $S_F$ и $S_{res}$:

In [16]:
S2_F = n1 * (fp_mean - height_mean) ** 2 + n2 * (hp_mean - height_mean) ** 2 + n3 * (lt_mean - height_mean) ** 2

S2_res = ((fp - fp_mean) ** 2).sum() + ((hp - hp_mean) ** 2).sum() + ((lt - lt_mean) ** 2).sum()

S2_F, S2_res

(253.9074675324678, 577.0568181818182)

Проверим выполнение равенства $S^2 = S_F^2 + S_{res}^2:$

In [19]:
S2_F + S2_res, ((height - height_mean) ** 2).sum()

(830.964285714286, 830.9642857142854)

Запишем оценки дисперсий:

In [20]:
k = 3
n = n1 + n2 + n3

sigma2_F = S2_F / (k - 1)
sigma2_res = S2_res / (n - k)

sigma2_F, sigma2_res

(126.9537337662339, 23.08227272727273)

Итак, значение статистики $T$:

In [21]:
T = sigma2_F / sigma2_res

T

5.500053450812598

Зафиксируем уровень значимости $\alpha = 0.05$. Для него найдём критическое значение $F_{crit}$:

In [23]:
alpha = 0.05

F_crit = stats.f.ppf(1 - alpha, (k - 1), (n - k))

F_crit

3.3851899614491687

Видим, что $T > F_{crit}$, поэтому заключаем, что отличие среднего роста спортсменом действительно является статистически значимым.

__Задача 2__

С помощью критерия Стьюдента для двух независимых выборок проверьте гипотезу о равенстве среднего роста футболистов и хоккеистов из задачи 1.

Рассмотрим статистику по критерию Стьюдента:
    
$$T = \dfrac{\overline{X_1} -\overline{X_2}}{\sqrt{\frac{\sigma_1^2}{n_1} + \frac{\sigma_2^2}{n_2}}}.$$

In [27]:
X1 = fp_mean

X1

179.125

In [29]:
X2 = hp_mean

X2

178.66666666666666

In [30]:
n1, n2 

(8, 9)

Несмещенная дисперсии для выборки соответственно:

In [39]:
sig_fp = fp.var(ddof=1)

sig_fp

16.982142857142858

In [40]:
sig_hp = hp.var(ddof=1)

sig_hp

28.499999999999996

Тогда статистика равна: 

In [73]:
T = (X1 - X2) / np.sqrt(sig_fp/n1 + sig_hp/n2)

T

0.19928601397363732

Число степеней свободы: 

In [74]:
f = n1 + n2 - 2 

f

15

Для уровня значимости $\alpha = 0.05$: 

Сравниваем полученное значение t-критерия Стьюдента 0.199 с критическим при р=0,05 значением, указанным в таблице: t_crit = 2.1314. 

Так как рассчитанное значение критерия больше критического, делаем вывод о том, что наблюдаемые различия статистически незначимы, гипотеза верна.